### ISSUE TITLE
Docker on Mac is very slow

### ISSUE THREAD

**Issue by dmuth** [docker/for-mac#5592]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: EC4E58BD-07DD-4312-897E-C4B9643C937C/20210416205423

### Expected behavior

Basic Docker commands like `docker ps` take less than a second to run when no containers are running.

### Actual behavior

```
$ time docker ps
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES

real    0m8.517s
user    0m0.157s
sys     0m0.166s
```

### Information

- Is it reproducible? - For me, yes.
- Is the problem new? - Yes.
- Did the problem appear with an update? - No.
- macOS Version: 10.14.5 (18F132)
- Intel chip or Apple chip: Intel.
- Docker Desktop Version: 3.3.1

### Steps to reproduce the behavior

Literally run any Docker command on the CLI.

Other things of note:

- Clicking on the Docker menu shows a spinning beachball for a few seconds as well as Docker does... something.
- Launching an image such as `alpine` now takes several seconds(!!), but once launched with an interactive shell, performance is normal.  For example, I can run the command `dd if=/dev/zero of=test bs=1M count=1000` and it takes 1-2 seconds which is normal for writing a GB of zeros to an SSD.
- I reset the app to its default settings
- I allocated 4 GB RAM to Docker
- CPU usage is very light, according to my widget from iStat Menus.

It's almost as if Docker is trying to talk to a network endpoint and then timing out.  I dunnno.  But I can't think of any way to further debug this on my end, hence my filing the ticket.

Thanks for reading,

-- Doug

**Comment by stephen-turner** [comment]
Well, I guess the first question is, do you have any corporate security software running that could be inserting itself on the network path? That would include the network connection between the host and the VM, it doesn't have to be going outside the host.

**Comment by dmuth** [comment]
@stephen-turner That's a good question!  So I tried a few more things this morning:

- Started up a CentOS box in Vagrant, installed docker on it, ran `docker ps` and `docker run hello-world`.  No issues. ✅ 
- Tried `docker run hello-world` on my Mac, took like 2.7 s. ❌ 

So it's weird that it didn't have the same issue in a VM on my Mac.  

To try and get more insight into network-related issues, I installed Little Snitch, rebooted (this was required), and tried to reproduce the problem.  Strangely, I was unable to!

Prior to this, my Mac had an uptime of about 60 days, so I'm wondering if there may have been some issue at the OS level that specifically affected Docker (or the underlying lightweight VM it uses) in a way that persisted across restarts of Docker, but a restart of the machine cleared up.

I'm afraid I don't have anymore insight into this--if the problem comes back, I can update this ticket of course.

-- Doug
